WordPress/gutenberg
# Date: merge 6.4 and 6.5 compat files

Resolves https://github.com/WordPress/gutenberg/issues/58431

## What and how?
This commit merges the 6.4 and 6.5 compat changes for `gutenberg_update_wp_date_settings()`, and stores them in a single place in compat/6.5.

### 6.4 compat changes (already [merged to Core for 6.4](https://github.com/WordPress/wordpress-develop/pull/5101))
- https://github.com/WordPress/gutenberg/pull/53931 
- https://github.com/WordPress/gutenberg/pull/58406

### 6.5 compat changes (needs backporting for 6.5)
- https://github.com/WordPress/gutenberg/pull/56682

## Why?

Compat code consolidation and to avoid duplicate filters.

## Testing Instructions
Fire up this branch and open the editor.

In the browser console, run `wp.date.getSettings()`.

Check that properties exist for:

1. `timezone.offsetFormatted` exists
2. `l10n.relative.*` exists

Smoke test functionality.

1. Follow test instructions in https://github.com/WordPress/gutenberg/issues/56254 to see timezone offset
2. Run `wp.date.humanTimeDiff( new Date() )` in the console. You should see a relative translation `"a second ago"`
